.github/workflows/cicd.yaml

```diff
@@ -0,0 +1,47 @@
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  buildAndPublish:
+    name: Build and publish to NPM (GH)
+    runs-on: ubuntu-22.04
+    permissions:
+      checks: write
+      contents: read
+      packages: write
+      statuses: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20
+          cache: 'yarn'
+          registry-url: https://npm.pkg.github.com
+      - name: Install dependencies
+        run: yarn --frozen-lockfile
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build
+        run: yarn build
```

Comment:
Can we add running tests before build + reporting results? If this is not possible, then we can remove permission for `checks` and `statuses`.

Reply:
Looks like tests here require credentials to some SaaS (or a local browser manually launched) to run.